### DISCUSSION TITLE
Nextjs project is very slow to save file in VS Code

### DISCUSSION THREAD

**Discussion by valcosmos** [vercel/next.js#38477] [Help] [answered]
I am building the project with nextjs and antd, in VS Code it is very slow every time I save the file.

I don't know why this is happening, does anyone know how to fix this?

<img width="1477" alt="image" src="https://user-images.githubusercontent.com/53572196/178110843-676b4dbe-6413-4991-a065-1a4a32d63c3f.png">

**Comment by RoobleCali** [comment]
i have the same issue also i hate page fast refresh

**Comment by valcosmos** [comment, marked as answer]
I have tried a lot in the past few days, and there seems to be no better solution. I found that using `WebStorm` will not have such a problem. Maybe using `WebStorm` is a good solution.

**Reply by ayush-dedhia25** [reply]
Same. Even I switched to webstorm for my next.js projects

**Reply by twinedo** [reply]
but Webstorm is not free

**Reply by Gasal246** [reply]
If you are using any of these extensions in the image just uninstall and restart your vscode, It worked for me.
..
![oVvDm](https://github.com/vercel/next.js/assets/104154453/04636eab-2d50-4429-9d73-f9626957ab4a)

**Reply by TrySpace** [reply]
Switching to Webstorm is not a solution, it's not even a workaround. 

It's like suggesting someone should move to another country because your neighbors have loud music.

**Reply by AaliyanAsif** [reply]
@Gasal246 solution is the right solution, worked like a charm for me.

**Reply by music-of-the-silence** [reply]
@Gasal246 Thanks a lot for the proper information. Hats off....!!

**Reply by revskill10** [reply]
> Switching to Webstorm is not a solution, it's not even a workaround.
> 
> It's like suggesting someone should move to another country because your neighbors have loud music.

You mean Next.js is the loud guy ?

**Reply by Garkatron** [reply]
Hero!

**Comment by CogitoBurrito** [comment]
Same here

**Comment by nosleepguy** [comment]
i have same issue. webstorm is not free

**Comment by hungnmhocmai** [comment]
I found the problem  with VSCode. It's Extention. I was disable GitLens, jest runner, live server extentions. All problem resolved!. I don't know exactly but it work ^^. I think is GitLens. I love VSCode, so i dont want to change WebStorm. Yaaa

**Reply by valcosmos** [reply]
Amazing 👍, and thank you for sharing, but I found that WebStorem is not bad😉.

**Reply by TrySpace** [reply]
Are you getting payed by Webstorm or something? Or are you just bitter?

**Reply by huyvu23** [reply]
> <img alt="" width="32" height="32" src="https://avatars.githubusercontent.com/u/102018491?s=64&amp;v=4">
> hungnmhocmai

thanks broo

**Comment by ayush-dedhia25** [comment]
So isn't there any good solution to get over that behavior?

**Reply by valcosmos** [reply]
Actually, I didn't find a good solution, but i am using WebStorm to avoid this situation.

**Reply by ayush-dedhia25** [reply]
Not sure if its due to any extension or vs code specific. I really hope that this issue resolves asap.

**Comment by aminyaghmaee** [comment]
delete your git lens and enjoy

**Reply by ayush-dedhia25** [reply]
If you mean to say GitLens extension then I don't have it installed 😅

**Reply by Karan-Dhingra** [reply]
Thank you It worked 🎉

**Comment by luga97** [comment]
I guess that maybe your problem is related to this issue https://github.com/prettier/prettier-vscode/issues/1333 and not with nextJS. btw my vs code is slow when i work with nextjs also, intellisense linter, everything is fucking me up, i don't know if maybe is a problem with linter and nextjs

**Reply by dondxniel** [reply]
I'm having the same issue; I'm working with Next 13. I've been having all round performance issues with compilation, load-up, file saving, everything.

**Comment by ayush-dedhia25** [comment]
I think creating a new vscode profile (from the profiles option) just for nextjs projects with only the extension required for the project development slightly fixes the issue.

**Comment by trungdvu** [comment]
Same here. So annoying. I disable all extensions except 'Vscode Vim' but it doesn't work until I disable Vim as well

**Comment by anthonyLock** [comment]
Hello, 

I had the same issue, for me I still had not excluded the `.next` folder in the `.prettierignore` Please see this for further details on this thread. I intially tried adding it to the `tsconfig` but it needs it for build step
https://github.com/prettier/prettier-vscode/issues/1333#issuecomment-757913931

Hope this helps as it is extremely annoying

**Comment by HamidBasri** [comment]
I had the same issue, my problem was because of the **auto import** extension, I added .next to the ignore path of this extension in the VSCode config and my problem solved.

**Reply by AhmedQasem4** [reply]
How to do that?

**Comment by gidgudgod** [comment]
I had the same issue with the Next JS 13 app stable version. The longer I run my app in dev mode the longer it takes for save, for intellicode to loads suggestions, also for prettier and eslint to auto format. Example case:

1. I stay and edits many times in one page.tsx file, which means nextjs will compile many times as well.
2. As time goes on, the slow loading of saving and formatting comes up.
3. The slow loading comes up more clear especially after I move to another file
4. In the end I tried to restart VSCode and run dev again.
5. At beginning after restart everything goes fine, but as time goes on... back to point number 1.

This is very strange bug because only in this Next JS 13 app project that the lags happen. Meanwhile I also opened other project like React project and Next JS 13 pages project, these projects run just fine.

For now still not yet found the solution, already tried to add .next in tsconfig and disabling extension, but lag still occurs.

**Comment by fossage** [comment]
I was running into the same issue and it turns out that it had to do with the TailwindCSS prettier plugin, not NextJS. By removing `prettier-plugin-tailwindcss` from my `prettier.config.js`, the project has been working smoothly for days now. You can follow the Github issue here: https://github.com/tailwindlabs/tailwindcss-intellisense/issues/755

**Reply by ahmed-adel2** [reply]
I don't have prettier, and it still happens..

**Comment by xitanggg** [comment]
I had this issue with `prettier-plugin-tailwindcss` version `0.3`, downgrading it back to say `0.2.1` with `npm install --save-dev prettier-plugin-tailwindcss@0.2.1`  and restarting VSCode solve the issue for me

**Comment by yishayhaz** [comment]
I think the best workaround is to create another VSC profile, within this profile do not install extensions such as gitlens, live server and so on (I'm not really sure why or what causes it)

**Comment by Cnerd-Mahadi** [comment]
just remove if you have any of these packages. mine is working fine after that.

![image](https://github.com/vercel/next.js/assets/52869547/6d3b61bd-75f7-4fc6-b325-448af261a651)

**Reply by DanielGardiner** [reply]
amazing! this fixed the issue for me 🥳

**Reply by Gasal246** [reply]
Thankyou Finally this solved my issue.. 🙂

**Reply by codeindevelop** [reply]
amazing! this fixed the issue for me 🥳

**Comment by TrySpace** [comment]
Probably the `Auto Import` extension issue is caused because the default setting: 

```json
"autoimport.filesToScan": "**/*.{js,jsx,ts,tsx,css}",
```

Which will by default scan all the files in `.next` and it ignores any other global ignore settings you try to set.

**Comment by AELONG2001** [comment]
Removing these two extensions from VS Code worked for me.

Auto Import
Auto Import - ES6, TS, JSX, TSX

**Comment by freitassdev** [comment]
My problem was that the disk on which the nextjs project was running was formatted in NTFS, which is not recommended for Ubuntu and which can also cause slowness, especially in cases like nextjs which performs several operations when you edit a file.

My solution was to format this disk to ext4, which is the default for Ubuntu and other Linux distros, this significantly improved performance in other applications and also fixed the problem with vscode taking a long time to update the file.
